modules/KalturaSupport/resources/mw.KCuePoints.js

```diff
@@ -383,11 +383,30 @@
 				filteredCuePoints = $.grep( filteredCuePoints, function ( cuePoint ) {
 					var foundCuePointType = _this.validateCuePointAttribute(cuePoint, "cuePointType", type);
 					var foundCuePointSubType = _this.validateCuePointAttribute(cuePoint, "subType", subType);
-					return foundCuePointType && foundCuePointSubType;
+					var checkCuePointsTag = _this.validateCuePointTags(cuePoint, _this.getPreviewCuePointTag());
+					return foundCuePointType && foundCuePointSubType && checkCuePointsTag;
 				} );
 			}
 			return filteredCuePoints;
 		},
+		//hide CP with tagName
+		validateCuePointTags: function(cuePoint, tagName){
+			var result  = cuePoint.tags.indexOf(tagName) === -1;
+			var playerConfig = this.embedPlayer.playerConfig;
+			if(playerConfig && playerConfig.plugins && playerConfig.plugins.dualScreen && playerConfig.plugins.dualScreen.allowAdminCuePoints){
+				result =  true;
+			}
+			return result;
+			
+		},
+		getPreviewCuePointTag:function () {
```

Comment:
@MaksymKorniiov to me this looks pretty static per player load. Shouldn't we try to do this once at load time instead of within every iteration of a filter?